shared/src/business/useCases/processStreamRecordsInteractor.test.js

```diff
@@ -669,4 +665,136 @@ describe('processStreamRecordsInteractor', () => {
       applicationContext.getSearchClient().bulk.mock.calls[0][0].body,
     ).toEqual([{ delete: { _id: '1_1', _index: 'efcms-case' } }]);
   });
+
+  it('calls delete function for failed records from bulkDelete', async () => {
+    applicationContext.getSearchClient().bulk.mockResolvedValue({
+      errors: [{ badError: true }],
+      items: [
+        {
+          delete: { error: false },
+        },
+        {
+          delete: { error: true },
+        },
+      ],
+    });
+
+    await processStreamRecordsInteractor({
+      applicationContext,
+      recordsToProcess: [
+        {
+          dynamodb: {
+            Keys: { pk: { S: '1' } },
+            OldImage: {
+              docketNumber: { S: '1' },
+              entityName: { S: 'Case' },
+              pk: { S: '1' },
+              sk: { S: '1' },
+            },
+          },
+          eventName: 'REMOVE',
+        },
+        {
+          dynamodb: {
+            Keys: { pk: { S: '2' } },
+            OldImage: {
+              docketNumber: { S: '2' },
+              entityName: { S: 'Case' },
+              pk: { S: '2' },
+              sk: { S: '2' },
+            },
+          },
+          eventName: 'REMOVE',
+        },
+      ],
+    });
+
+    expect(applicationContext.getSearchClient().delete).toBeCalledTimes(1);
+    expect(
+      applicationContext.getSearchClient().delete.mock.calls[0][0],
+    ).toMatchObject({ id: '2_2', index: 'efcms-case' });
+  });
+
+  it('logs error if bulk delete throws an error', async () => {
+    const error = new Error('bad!');
+    applicationContext.getSearchClient().bulk.mockImplementation(() => {
+      throw error;
+    });
+
+    await processStreamRecordsInteractor({
+      applicationContext,
+      recordsToProcess: [
+        {
+          dynamodb: {
+            Keys: { pk: { S: '1' } },
+            OldImage: {
+              docketNumber: { S: '1' },
+              entityName: { S: 'Case' },
+              pk: { S: '1' },
+              sk: { S: '1' },
+            },
+          },
+          eventName: 'REMOVE',
+        },
+      ],
+    });
+
+    expect(applicationContext.notifyHoneybadger).toBeCalledTimes(1);
+    expect(applicationContext.notifyHoneybadger.mock.calls[0][0]).toEqual(
```

Comment:
@ericsorenson This wouldn't work in the court's environment since it doesn't use Honeybadger, right?

Reply:
Errors won't be sent to Honeybadger unless Honeybadger is configured. These tests only use a mocked API.

Reply:
I wonder if we need to audit if any of the items being sent to Honeybadger also need to be logged to Cloudwatch? Maybe you already know the answer to this, but this is something that could be followed up on.

Reply:
todo for me: add auditing Honeybadger error logs to the task list for log auditing

Reply:
They're logged in both places:
```
applicationContext.logger.info('Error', e);
await applicationContext.notifyHoneybadger(e);
```